Shopify/cli
# Add support for version api version

### WHY are these changes introduced?
Follow-up: https://github.com/Shopify/cli/pull/3226

- Add support for the `webhooks -> api_version` field included in this [PR](https://github.com/Shopify/shopify/pull/472181)
- As a side effect of adding `declarative webhooks` support without `api_version`, `webhooks` field always appears as `new` in the confirmation prompt even if no changes have been made. Check the `NOTE` comment added to the tophat instructions of this [PR](https://github.com/Shopify/cli/pull/3205) for more details


### WHAT is this pull request doing?
- Add support for the `webhooks -> api_version` field included in this [PR](https://github.com/Shopify/shopify/pull/472181)
- As now the `webhooks` section can be returned without subscriptions they should be handled as optional when the content is received as 


### How to test your changes?
- Create a new `cli` app and run the `p shopify app config link --path /PATH/TO/YOUR/APP` command to link it to a new remote app. `webhooks -> api_version` is populated inside the `toml` file
- Run `SHOPIFY_CLI_VERSIONED_APP_CONFIG=1 npm run shopify app deploy --path /PATH/TO/YOUR/APP`  and confirm
<img src="https://github.com/Shopify/cli/assets/105213827/fcd98c63-f5e6-47e2-a336-56e945cbe4ec" width=300/>

- The first time you run the `deploy` command with `version config` enabled all the config modules are new despite the app has those values in the `api client configuration``
- Modify the `api_version` value to `unstable` and run `SHOPIFY_CLI_VERSIONED_APP_CONFIG=1 npm run shopify app deploy --path /PATH/TO/YOUR/APP` again and confirm
<img src="https://github.com/Shopify/cli/assets/105213827/fd651c35-87c0-42c7-bf1a-f40d42b63386" width=300/>
- Now `webhooks` is not displayed as new
- Modify the `api_version` again to a different value and run `p shopify app config link --path /PATH/TO/YOUR/APP`. The value should be overwritten with the remote one




### Post-release steps



### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
